app/src/main/java/com/tolkiana/tacoapp/utilities/ApplicationImageLoader.kt

```diff
@@ -0,0 +1,21 @@
+package com.tolkiana.tacoapp.utilities
+
+import android.content.Context
+import com.android.volley.toolbox.ImageLoader
+import com.android.volley.toolbox.Volley
+
+/**
+ * Created by tolkiana on 9/17/17.
+ */
+class ApplicationImageLoader {
+    companion object {
+
+        @Volatile private var INSTANCE: ImageLoader? = null
+
+        fun getInstance(context: Context): ImageLoader = INSTANCE ?: synchronized(this) {
+            INSTANCE ?: buildImageLoader(context.applicationContext).also { INSTANCE = it }
```

Comment:
I actually have another question, I'm not sure I understand this line:
`fun getInstance(context: Context): ImageLoader = INSTANCE ?: synchronized(this)`
I think I'm I little confused by the syntax. I understand when you have only one line in the function you can omit the braces like what we have for buildImageLoader but here I'm a little bit lost.

I think I understand the second line
`INSTANCE ?: buildImageLoader(context.applicationContext).also { INSTANCE = it }`
which I believe is saying if the object exist just get that otherwise create it and assign it.
Also I don't see areturn is that something we can do just in certain kind of functions?